receiver/hostmetricsreceiver/internal/scraper/networkscraper/network_scraper.go

```diff
@@ -160,7 +160,15 @@ func (s *scraper) recordNetworkConnectionsMetrics() error {
 
 	tcpConnectionStatusCounts := getTCPConnectionStatusCounts(connections)
 
-	s.recordNetworkConnectionsMetric(now, tcpConnectionStatusCounts)
+	s.recordNetworkTCPConnectionsMetric(now, tcpConnectionStatusCounts)
+
+	connections, err = s.connections("udp")
+	if err != nil {
+		return fmt.Errorf("failed to read UDP connections: %w", err)
+	}
+
+	// TODO: state attribute should be omitted for UDP connections
```

Comment:
I understand your second question now. I answered before looking at the code. This seems like a good reason to separate metrics. Attribute values should not be dependent on each other. Can you please bring this problem to the spec issue for separating the metric. Maybe we should wait until that spec issue is resolved before merging this?

Reply:
👍🏻 I'll see what the feedback from the spec to my question in this PR is https://github.com/open-telemetry/opentelemetry-specification/pull/2617, please add yours there. I can open a separate issue if enough folks feel that separating the protocol makes sense.

Reply:
Just chiming in to say that I agree that the initial approach of recording these as separate metrics makes the most sense.